salt/state.py

```diff
@@ -2507,22 +2508,16 @@ def _get_envs(self):
             envs.extend([x for x in list(self.opts['file_roots'])
                          if x not in envs])
         env_order = self.opts.get('env_order', [])
+        # Remove duplicates
+        env_order = list(collections.OrderedDict.fromkeys(env_order))
```

Comment:
There are multiple variants to remove duplicates:
```python
>>> prep = 'env_order = "[\'foo\', \'bar\', \'foo\', \'baz\']"; import collections'
>>> import timeit
>>> t1=timeit.Timer('sorted(set(env_order), key=lambda x: env_order.index(x))', prep)
>>> t1.repeat()
[4.158413887023926, 3.958709955215454, 3.9590039253234863]
>>> t1 = timeit.Timer('list(set(env_order))', prep)
>>> t1.repeat()
[0.9367499351501465, 0.903162956237793, 0.9018638134002686]
>>> t2 = timeit.Timer('list(collections.OrderedDict.fromkeys(env_order))', prep)
>>> t2.repeat()
[16.622393131256104, 16.300525903701782, 16.367045164108276]
>>> t3=timeit.Timer('sorted(set(env_order), key=lambda x: env_order.index(x))', prep)
>>> t3.repeat()
[4.10307502746582, 4.041788101196289, 4.156888961791992]
>>> t4=timeit.Timer('[v for (i, v) in enumerate(env_order) if v not in env_order[0:i]]', prep)
>>> t4.repeat()
[2.9662280082702637, 2.8464999198913574, 2.7960379123687744]
>>> t5=timeit.Timer('dummy = set(); [(v, dummy.add(v))[0] for v in env_order if v not in dummy]', prep)
>>> t5.repeat()
[2.429576873779297, 2.391568899154663, 2.5268590450286865]
```
Which of the order preserving variants do you prefer?